syntax = "proto3";

package flyte.plugins.sagemaker;

option go_package = "github.com/kumare3/awsflyteplugins/gen/pb-go/proto";

message AlgorithmSpecification {
  string TrainingImage = 1;

  enum TrainingInputModeEnum {
    Pipe = 0;
    File = 1;
  }
  TrainingInputModeEnum TrainingInputMode = 2;
  string AlgorithmName = 3;

  message MetricDefinition {
    string Name = 1;
    string Regex = 2;
  }
  repeated MetricDefinition MetricDefinitions = 4;
}

message ResourceConfig {
  string InstanceType = 1; 
  int32 InstanceCount = 2;
  int32 VolumeSizeInGB = 3;
  string VolumeKmsKeyId = 4;
}

message StoppingCondition {
  int32 MaxRuntimeInSeconds = 1;
  int32 MaxWaitTimeInSeconds = 2;
}

message VpcConfig {
  repeated string SecurityGroupIds = 1;
  repeated string Subnets = 2;
}

message SagemakerHPOJob {
  string RoleArn = 1;
  AlgorithmSpecification AlgorithmSpecification = 2;
  ResourceConfig ResourceConfig = 3;
  StoppingCondition StoppingCondition = 4;
  VpcConfig VpcConfig = 5;
  bool EnableSpotTraining = 6;
}
